api/src/main/java/org/openmrs/module/ugandaemrreports/data/converter/IYCFDataConverter.java

```diff
@@ -23,20 +23,18 @@
 public class IYCFDataConverter implements DataConverter{
     @Override
     public Object convert(Object obj) {
-
-        if (obj == null) {
-            return "";
-        }
         Concept iyfcResults = ((Obs) obj).getValueCoded();
-
-        if(iyfcResults != null && iyfcResults.equals(Dictionary.getConcept(Dictionary.YES_CIEL))){
-            return "Y";
+        if(iyfcResults==null)
+        {
+            return null;
         }
-
-        else if(iyfcResults != null && iyfcResults.equals(Dictionary.getConcept("1066AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"))){
-            return "N";
+        if (iyfcResults.equals(Dictionary.getConcept("dcd695dc-30ab-102d-86b0-7a5022ba4115"))) {
+                return "Y";
         }
-        return null;
+        if (iyfcResults.equals(Dictionary.getConcept("dcd69c06-30ab-102d-86b0-7a5022ba4115"))) {
+                return "N";
+            }
+        return ((Obs) obj).getValueCoded();
```

Comment:
@solemabrothers This should be return patientProgramsToUpdate.getValueCoded() as the method is only applicable to concepts

Reply:
@ssmusoke i was of the view that it should return `iyfcResults` provided that none of the  conditions above are met